plugins/redis/redis.cpp

```diff
@@ -749,8 +749,8 @@ void Connection::handleLockOnGet(ICodeContext * ctx, const char * key, MemoryAtt
     //Requires separate connection from GET so that the replies are not mangled. This could be averted
     Owned<Connection> subConnection = new Connection(ctx, options.str(), ip.str(), port, serverIpPortPasswordHash, database, password, timeout);
     OwnedReply reply = Reply::createReply(redisCommand(subConnection->context, "SUBSCRIBE %b", channel.str(), (size_t)channel.length()));
-    assertOnCommandErrorWithKey(reply->query(), "GET", key);
-    if (reply->query()->type == REDIS_REPLY_ARRAY && strcmp("subscribe", reply->query()->element[0]->str) != 0 )
+    subConnection->assertOnCommandErrorWithKey(reply->query(), "GET", key);
+    if (reply->query()->type != REDIS_REPLY_ARRAY || strcmp("subscribe", reply->query()->element[0]->str) != 0 )
```

Comment:
In 6.0 I move this entire test till just before the read of the subscription as we don't care if this failed if it's not actually needed, i.e. the key wasn't locked.